python_modules/libraries/dagster-aws/dagster_aws_tests/s3_tests/test_s3_sensor.py

```diff
@@ -59,6 +63,69 @@ def put_key_in_bucket(key, body):
         # The test would be flaky if sorted was not guaranteed to be stable, which it is
         # according to: https://wiki.python.org/moin/HowTo/Sorting/.
         keys = get_s3_keys(
-            bucket=bucket_name, prefix=prefix, since_key=thousand_and_one_hundred_keys[1090]
+            bucket=bucket_name,
+            prefix=prefix,
+            since_key=thousand_and_one_hundred_keys[1090],
         )
         assert len(keys) == 9, "9 keys should be returned"
+
+
+def test_get_s3_keys_missing_since_key():
+    bucket_name = "test-bucket"
+    prefix = "content"
+
+    with moto.mock_s3():
+        s3_client = boto3.client("s3", region_name="us-east-1")
+        response = s3_client.create_bucket(Bucket=bucket_name)
+        if not response:
+            raise Exception("Failed to create bucket")
+
+        def put_key_in_bucket(key, body):
+            r = s3_client.put_object(Bucket=bucket_name, Key=f"{prefix}/{key}", Body=body)
+            if not r:
+                raise Exception("Failed to create object")
+
+        no_key = get_s3_keys(bucket=bucket_name, prefix=prefix)
+        assert len(no_key) == 0  # no keys in bucket
+
+        put_key_in_bucket(key="foo-1", body="test")
+        put_key_in_bucket(key="foo-2", body="test")
+        two_keys = get_s3_keys(bucket=bucket_name, prefix=prefix)
+        assert len(two_keys) == 2
+
+        # if `since_key` is not present, returns all object keys
+        missing = get_s3_keys(bucket=bucket_name, prefix=prefix, since_key="content/foo-z")
+        assert len(missing) == 2
```

Comment:
Let me know if we would prefer to throw an exception if the key is not present.

Reply:
Yep I think we should do that